### DISCUSSION TITLE
Please integrate basujindal's turbo mode and vram optimization to hlky web UI

### DISCUSSION THREAD

**Discussion by mhnoni** [hlky/stable-diffusion-webui#910] [Feature suggestions]
Basujindal turbo should increase Vram by 1GB but it makes it faster. 
And basujindal ram optimization will split the vram when needed to render a bigger image. 
both these feature are available with Basujindal fork but has no UI like hlky.

Edit: nmv, I just tried basujindal fork, and the turbo is faster than the optimized slow not against unoptimized fast mode, he should've made it more clear to avoid confusion.

Thanks!

**Comment by Michoko92** [comment]
Those modes are already available and can be activated in the options, inside the relauncher.py script.

**Reply by mhnoni** [reply]
> Those modes are already available and can be activated in the options, inside the relauncher.py script.

Can you tell us how to activate each? I don't think I found a guide so far explaining how to activate turbo mode.

**Reply by AscendedGravity** [reply]
In the scripts folder, edit relauncher.py. You'll see different options at the top. Change False to True, save the changes, and launch.

**Reply by mhnoni** [reply]
> In the scripts folder, edit relauncher.py. You'll see different options at the top. Change False to True, save the changes, and launch.

oh, thanks a lot, looks like this is a new feature added later since my relauncher.py didn't have it, I'm going to update mine right now.

**Reply by mhnoni** [reply]
> Those modes are already available and can be activated in the options, inside the relauncher.py script.

Ok, so my GTX 1070 8GB takes 37 sec and 6.2GB in a normal mode, which I assume its not optimized.

I just tried both below features from your latest fork:
- optimized = True (this made cut the vram to 4.3GB and made it slower) as expected and took above 1.2min to finish the render. 
- optimized_turbo = True (This cut the Vram  to a similar size 4.3 GB) but added a few sec to finish the render ~ 40 sec 

So I'm going to assume these both are Vram optimization and not the feature I asked -> "basujindal's turbo mode" which adds 1 GB VRam but makes the render faster than the normal case. Can you please check and let me know If I'm mistaken here? 
Or basujindal's turbo mode is basically optimized_turbo and it doesn't make it faster than the normal speed but faster than the optimized?
May you please explain what optimized_turbo actually does under the hood?

**Reply by leonnn1** [reply]
I tested and on my PC it worked fine, the Optmized decreased the memory consumption but it was extremelly slow
On the optmized turbo it decreased the memory consumption and it was almost as fast as the regular one.
About add 1 gb of ram and make stuff even faster I couldn't find anything related

**Reply by mhnoni** [reply]
> About add 1 gb of ram and make stuff even faster I couldn't find anything related

that's on the basujindal fork. 

https://github.com/CompVis/stable-diffusion/commit/6eee78cf67388b2076b851c5d31140dddb4d8718#diff-bcd58f5c9e6b9e90021dfe4292a9fe6254234aa551b559ef2b853aa1800ed1dc

Edit: nmv, I just tried basujindal fork, and the turbo is faster than the optimized slow not against unoptimized fast mode, he should've made it more clear to avoid confusion.
